### PR TITLE
HTTP2 support

### PR DESCRIPTION
**Pull request by vkarpov15**
**What this PR does**:

Subs out axios for http2 using examples from [this blog post](https://www.sohamkamani.com/nodejs/http2/). Tests pass locally, and http2 seems to be quicker than axios as well.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by kathirsvn** [comment]
Hi @vkarpov15 do you think this PR is ready for review / merge now?

**Comment by vkarpov15** [comment]
@kathirsvn this PR is ready for review :+1:

**Comment by vkarpov15** [comment]
I added a `useHTTP2` option that we can enable/disable, so this PR should be fairly low risk

**Comment by jeffreyscarpenter** [comment]
@kathirsvn are you going to review?

**Comment by kathirsvn** [comment]
> @kathirsvn are you going to review?

Yes, we discussed about this and planned to review and get this completed this week.

File: src/client/httpClient.ts [hunk 1]
```diff
@@ -45,6 +46,7 @@ interface APIClientOptions {
   authUrl?: string;
   isAstra?: boolean;
   logSkippedOptions?: boolean;
+  useHTTP2?: boolean;
```
Comment (kathirsvn) [inline src/client/httpClient.ts:49]: Can we make this as the default option?

**Review by kathirsvn — APPROVED** [review]
LGTM

File: src/client/httpClient.ts (hunk 1 above)
Comment (vkarpov15) [reply src/client/httpClient.ts:49]: @kathirsvn thanks for review. Do you mean make `useHTTP2` true by default?

**Comment by jeffreyscarpenter** [comment]
@vkarpov15 ok to edit the PR title to remove "Proof of concept"?

**Comment by vkarpov15** [comment]
@jeffreyscarpenter done

**Pull request merged**